guides/localized-message-source/localized-message-source.adoc

```diff
@@ -0,0 +1,49 @@
+common:header.adoc[]
+
+common:requirements.adoc[]
+
+common:completesolution.adoc[]
+
+common:create-app-features.adoc[]
+
+=== Properties
+
+Create a default `messages.properties` file:
+
+resource:i18n/messages.properties[]
+
+Create a `messages_es.properties` file for Spanish locale:
+
+resource:i18n/messages_es.properties[]
+
+=== Message Source
+
+Create a `MessageSource` that uses the previous properties files:
+
+source:MessageSourceFactory[]
+
+callout:factory[1]
+callout:singleton[2]
+
+=== Controller
+
+source:HelloWorldController[]
+
+callout:controller[number=1,arg0=/]
+callout:constructor-di[number=2,arg0=LocalizedMessageSource]
+callout:produces[3]
+<4> https://docs.micronaut.io/latest/api/io/micronaut/context/LocalizedMessageSource.html[LocalizedMessageSource] contains methods to fetch a message with a code, a default message, or variables.
+
+=== Tests
+
+test:HelloWorldControllerTest[]
+
+callout:micronaut-test[1]
+callout:http-client[2]
+callout:http-request[3]
+
+common:testApp.adoc[]
+
+== Learn More
+
+The application changes the locale with `Accept-Language` HTTP Header. Learn more about https://docs.micronaut.io/latest/guide/#localeResolution[Locale Resolution].
```

Comment:
I think this is missing an article, but I'm not sure which is more appropriate.

```suggestion
The application changes the locale with an `Accept-Language` HTTP Header. Learn more about https://docs.micronaut.io/latest/guide/#localeResolution[Locale Resolution].
```

OR

```suggestion
The application changes the locale with the `Accept-Language` HTTP Header. Learn more about https://docs.micronaut.io/latest/guide/#localeResolution[Locale Resolution].
```